cache/models/samples.js

```diff
@@ -452,6 +457,309 @@ function upsertOneSample(sampleQueryBodyObj, method, isBulk) {
 module.exports = {
 
   /**
+   * Delete sample. Get sample. If found, get aspect, delete sample entry from
+   * sample index, delete aspect from subject set and delete sample hash. If
+   * sample not found, throw Not found error.
+   * @param  {String} sampleName - Sample name
+   * @param  {Object} logObject - Log object
+   * @param  {String} method - Type of request method
+   * @returns {Promise} - Resolves to a sample object
+   */
+  deleteSample(sampleName, logObject, method) {
+    const subjAspArr = sampleName.toLowerCase().split('|');
+
+    if (subjAspArr.length < TWO) {
+      const err = new redisErrors.ValidationError({
+        explanation: 'Incorrect sample name.',
+      });
+      return Promise.resolve(err);
+    }
+
+    const subjAbsPath = subjAspArr[ZERO];
+    const aspName = subjAspArr[ONE];
+    const cmds = [];
+    let sampObjToReturn;
+
+    return redisOps.getHashPromise(sampleType, sampleName)
+    .then((sampleObj) => {
+      if (!sampleObj) {
+        throw new redisErrors.ResourceNotFoundError({
+          explanation: 'Sample not found.',
+        });
+      }
+
+      sampObjToReturn = sampleObj;
+    })
+    .then(() => {
+      cmds.push(redisOps.getHashCmd(aspectType, aspName));
+      cmds.push(redisOps.delKeyFromIndexCmd(sampleType, sampleName));
+      cmds.push(redisOps.delAspFromSubjSetCmd(subjAbsPath, aspName));
+      cmds.push(redisOps.delHashCmd(sampleType, sampleName));
+
+      return redisOps.executeBatchCmds(cmds);
+    })
+    .then((response) => {
+      logObject.dbTime = new Date() - logObject.reqStartTime; // log db time
+      const asp = response[ZERO];
+
+      // attach aspect and links to sample
+      const resSampAsp = cleanAddAspectToSample(sampObjToReturn, asp, method);
+      resSampAsp.isDeleted = new Date().toString();
```

Comment:
Correct. And also FYI, whenever I need to set the date, I use new `Date().toString()`, which resolves to `Wed Mar 15 2017 14:58:53 GMT-0700 (PDT)`.